### PR TITLE
Construct World

### PR DESCRIPTION
**Pull request by tsukiy0**
* From 2D array (simple world)
* Update World internals to use Map<Coordinate, Cell> to better represent a 2D grid in my mind
* Add the Cell domain object
* Test World#isAlive

### CODE REVIEW

File: src/main/java/io/tsukiyo/gameoflife/core/Game.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
     private World world;
 
     public void initialize(boolean[][] simpleWorld) {
-        world = new World();
+        world = new World(simpleWorld);
```
Comment (mwpeng) [inline src/main/java/io/tsukiyo/gameoflife/core/Game.java]: is  simpleWorld still a world? 
does the name give us information for understanding what are the values in the 2D array?

File: src/main/java/io/tsukiyo/gameoflife/core/World.java [hunk 2]
```diff
@@ -1,7 +1,56 @@
...
 
+import java.util.HashMap;
+import java.util.Map;
+
 public class World {
+    private Map<Coordinate, Cell> store;
+    private Integer height;
+    private Integer width;
+
+    public World(boolean[][] simpleWorld) {
```
Comment (mwpeng) [inline src/main/java/io/tsukiyo/gameoflife/core/World.java]: Is this 2D array parameter a world? 
I would assume even a simpleWorld would contain certain logic/behaviour.
It is more like to be the the initial state  that a world starts with.

File: src/main/java/io/tsukiyo/gameoflife/core/World.java [hunk 3]
```diff
@@ -1,7 +1,56 @@
...
+    private Integer heightFromSimpleWorld(boolean[][] simpleWorld) {
+        return simpleWorld.length;
+    }
+
     public boolean isAlive() {
-        return true;
+        return store.entrySet().stream().anyMatch(entry -> entry.getValue().isAlive());
+    }
+
+    public boolean[][] toSimpleWorld() {
```
Comment (mwpeng) [inline src/main/java/io/tsukiyo/gameoflife/core/World.java]: ToSomething  indicates the class is converted to another type, Like:
Integer.toString(int)  
Is SimpleWold a type/class  or a representation?
What is the purpose of this method? converting to a different type or retrieving a representation in the format of 2D array?

File: src/main/java/io/tsukiyo/gameoflife/core/World.java (hunk 2 above)
Comment (tsukiy0) [reply src/main/java/io/tsukiyo/gameoflife/core/World.java]: how about `worldSnapshot`? still a world, but captured at a point in time for sharing

Comment (mwpeng) [reply src/main/java/io/tsukiyo/gameoflife/core/World.java]: better than simpleWorld

**Comment by mwpeng** [comment]
That sounds good for me.

**Pull request merged**